pkg/kvstore/etcd.go

```diff
@@ -508,6 +563,15 @@ func connectEtcdClient(config *client.Config, cfgPath string, errChan chan error
 		},
 	)
 
+	ec.controllers.UpdateController("kvstore-etcd-lock-session-renew",
+		controller.ControllerParams{
+			DoFunc: func(ctx context.Context) error {
+				return ec.renewLockSession()
+			},
+			RunInterval: time.Duration(10) * time.Millisecond,
```

Comment:
This is quite frequent - does this put any stress onto the kvstore?

Reply:
>This is quite frequent - does this put any stress onto the kvstore?

It's not frequent, the `ec.renewLockSession()` blocks on a channel.